feat(cli): support multiple output formats via --formats flag

JuanVqz opened the pull request:
# Context
We added support to `JSON`, and `HTML` reports.

# Done When
We can generate those reports with the `--formats` flag
```
skunk . --formats console,json,html # if you need the 3 format reports
skunk app lib --formats json # will only save the json result at `tmp/rubycritic/skunk_report.json`
skunk app # will use console format by default
```

JuanVqz committed:
chore: update rubocop todo configuration

Update rubocop todo file with latest offense counts and configurations

JuanVqz commented on CHANGELOG.md:
```diff
@@ -7,6 +7,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## main [(unreleased)](https://github.com/fastruby/skunk/compare/v0.5.4...HEAD)
 
+* [FEATURE: Add `--formats` CLI flag to select report formats (json, html, console)]
```
```suggestion
* [FEATURE: Add `--formats` CLI flag to select report formats (json, html, console)](https://github.com/fastruby/skunk/pull/130)
```

JuanVqz committed:
Apply suggestions from code review

etagwerker commented on test/lib/skunk/commands/help_test.rb:
```diff
@@ -12,6 +12,7 @@
         Usage: skunk [options] [paths]
             -b, --branch BRANCH              Set branch to compare
             -o, --out FILE                   Output report to file
+            -f, --formats json,html,console  Output formats: json,html,console
```
```suggestion
            -f, --formats json,html,console  Output formats: json,html,console (default: console)
```

etagwerker commented on lib/skunk/cli/options/argv.rb:
```diff
@@ -26,6 +27,10 @@ def parse
               self.output_filename = filename
             end
 
+            opts.on("-f", "--formats json,html,console", Array, "Output formats: json,html,console") do |list|
```
Any way to tell that the default is console here?

etagwerker reviewed (commented):
@JuanVqz Changes look good, I just added one comment related to default

JuanVqz committed:
fix(cli): update help text to show default output format

Add test case to verify default console format when no --formats option is provided

etagwerker reviewed (changes requested):
@JuanVqz Unfortunately now I'm getting non-deterministic failures from the test suite and it seems to be related to these changes.

https://gist.github.com/etagwerker/9ac9d8e20b5f5a4789b5fe263946c211

1 or 2 out of 10 test executions will fail with something like that.

Could you please take a look?

JuanVqz committed:
fix(cli): wrap formats option

JuanVqz committed:
fix(cli): reset config before parse

JuanVqz committed:
fix(deps): pin path_expander (<2.0) for Ruby 2.7 (endless method defs)

JuanVqz commented on lib/skunk/cli/application.rb:
```diff
@@ -25,6 +25,7 @@ def initialize(argv)
       def execute
         warn_coverage_info unless File.exist?(COVERAGE_FILE)
 
+        Skunk::Config.reset
```
> https://gist.github.com/etagwerker/9ac9d8e20b5f5a4789b5fe263946c211
> 1 or 2 out of 10 test executions will fail with something like that.

Fix for ^

Resetting the `format` key to console fixed it.

JuanVqz commented on skunk.gemspec:
```diff
@@ -38,6 +38,7 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ["lib"]
 
+  spec.add_dependency "path_expander", "< 2.0"
```
> Unfortunately now I'm getting non-deterministic failures from the test suite, and it seems to be related to these changes.

Fix for ^

`path_expander` is a transitive dependency of `flay`, which is used by `rubycritic`, which `Skunk` depends on. So even though you don’t require it directly, it ends up in your bundle via `rubycritic -> flay -> path_expander`, and this is because rubycritic now only supports Ruby 3+
